public/js/lockToggle.js

```diff
@@ -0,0 +1,25 @@
+
+
+
+const urlParams = new URLSearchParams(window.location.search);
+const urlID = urlParams.get('id')
+ document.querySelectorAll('#lockToggle').forEach(toggleLock => {
+        toggleLock.addEventListener('click',(e) => {
+            const fieldName = toggleLock.dataset.field
+            const URL = `http://localhost:8080 / api / lockApplicantField ? id = ${urlID} & field = ${fieldName}`
```

Comment:
why the spaces throughout the URL? also, dont use localhost:8080, just use `/api/lockApplicantField...`